hack/update-staging-godeps-dockerized.sh

```diff
@@ -103,20 +103,24 @@ function diffGodepManifest() {
   local ret=0
   diff --ignore-matching-lines='^\s*\"GoVersion\":' --ignore-matching-line='^\s*\"GodepVersion\":' --ignore-matching-lines='^\s*\"Comment\"' -u "${KUBE_ROOT}/staging/src/k8s.io/${repo}/Godeps/Godeps.json" "${TMP_GOPATH}/src/k8s.io/${repo}/Godeps/Godeps.json" || ret=$?
   if [[ "${ret}" != "0" && "${FAIL_ON_DIFF}" == true ]]; then
-    exit ${ret}
+    return ${ret}
```

Comment:
What's the intent of these changes? We don't want fast fail?

Reply:
We have so many, we want to report all of the failures at once, rather than force someone to have to go one by one.

Reply:
Basically if you change anything in apimachinery, you have to fix 10 different staging godeps.

Reply:
Are you talking locally, or in CI? Looking at testgrid, this verify test takes approx 15 minutes to complete.. but fast failing cuts that in about half, if this particular test step is going to fail. I'm not 100% sure it's worth reporting out all the failures, when if one if them fails, the "hint" should be "run `hack/update-staging-godeps.sh` to fix all of them".

Reply:
In CI.

Running hack/update-staging-godeps sucks locally if you are on Mac.  As in, is unbelievably painful.  It requires docker, volume sharing, a very specific config, etc.  If you don't have an env to recreate, you end up running far more test iterations against CI (by fixing, pushing, fixing, pushing) and far more test runs than failing fast saves.

Reply:
Docker and volume sharing, yes. A very specific config, not really... that was our point in containerizing it, is so that you don't have to have a very specific godeps setup to do this (godeps is tempermental, but that's another topic). I use Docker for Mac on my machine, and run this all the time.

The [documented](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/godep.md#saving-deps-in-staging-repos) way updating staging godeps is to use the update script locally. Trying to use CI (by fixing, pushing, fixing, pushing) is not supported.

If you'd like to continue discussing this separately, I think we should pull it into another issue/PR, rather than tag this onto an unrelated change.

Reply:
Fine - this is one of my biggest personal frustrations as a maintainer on the core staging repos, but it does deserve a different PR.  The point of the PR infrastructure is to take burdens off programmers, and forcing someone who wants a machine to run this check for them to step through seems to defeat that goal.

Reply:
Moved to https://github.com/kubernetes/kubernetes/pull/74087